### PR TITLE
fix(suite-desktop): workaround desktop bundle size by hiding react-native related dependencies

### PR DESCRIPTION
**Pull request by matejkriz**
🏳️ I'm almost giving up, if anyone would like to try to make dependency bundling work without duplication and with working native dependencies, I'd welcome help...

This is here prepared as a workaround to avoid a blocked release.

### CODE REVIEW

**Review by mroz22 — APPROVED** [review]
feels acceptable. does it bring any possible problems?

**Comment by matejkriz** [comment]
> feels acceptable. does it bring any possible problems?

Theoretically in some cases it can confuse NX and some tests or builds that should be run will not run. For example if we update react-native maybe?

**Comment by matejkriz** [comment]
Maybe I found a solution that allows desktop build, NX and eslint to all have their cake and eat it too! 

If I put those dependencies as optional peer dependencies (because they really are, they're only needed in a certain context of use) then they are not copied to desktop build, esbuild is satisfied and NX can resolve the dependency tree 🎉

**Comment by matejkriz** [comment]
I just tested the Nx - 

with optional peerDependency:
1. added `@trezor/device-utils` as optional peerdependency of `@trezor/env-utils`
1. run `nx:type-check` to cache results
1. change something in device-utils
1. run `nx:type-check` again
1. `@trezor/env-utils` correctly rebuilded and type check evaluated again, other packages use local cache


without optional peerDependency:
1. removed optional peerdependency
1. run `nx:type-check` to cache results
1. change something in device-utils
1. run `nx:type-check` again
1. `@trezor/env-utils` correctly use local cache and not rebuilded

✅

**Comment by szymonlesisz** [comment]
i''l just put my two cents here :)

i don't think `react-native` should be a dependency of connect at all.

1. we are forcing all 3rd parties to install it (or will throw warnings about missing peerDependencies?)
2. it's used **only** by us (`@trezor/transport-native` is not a public/published package)
3. afaik currently it's **not used** even by us :)

I remember few months back when we talk about it with @mroz22 while he was refactoring transport package that architectonically correct would be to:

- `@trezor/connect` `init` to accept any `Transport` class interface (it's already there by the types)
- move `ReactNativePlugin`file to `@trezor/transport-native` package (and maybe publish it all together?)
- initialize connect in `suite-native` with transport set to `ReactNativePlugin` (which probably should be renamed to ReactNativeTransport?)

**TLDR;**

i would remove this dependency from connect (commit before the fixup) to make sure it would not be published (luckily it wasn't yet) and left eslint-disabled with the comment.

**Comment by matejkriz** [comment]
No, react-native is marked as optional via `peerDependenciesMeta` supported by both [NPM](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependenciesmeta) and [Yarn](https://yarnpkg.com/configuration/manifest#peerDependenciesMeta).  

> Marking a peer dependency as optional ensures npm will not emit a warning 

I think `react-native` currently is dependency of connect and the discussion is only whether to admit it or not in package.json... Injecting ReactNativePlugin sounds great. Optional peer dependency strikes me as a good temporary solution.

**Comment by szymonlesisz** [comment]
i'm pretty sure that it is not in latest released version :)

and as a proof i'm including yarn.lock of freshly created project with only one dependency `@trezor/connect version 9.0.11`

[yarn.lock.txt](https://github.com/trezor/trezor-suite/files/11919933/yarn.lock.txt)

it was added as dependency 9 days ago to [fix extraneous dependencies](https://github.com/trezor/trezor-suite/commit/6707f7df93272be0c8b54d02c9b847c8ec4bd483)

**Comment by szymonlesisz** [comment]
and im still pretty convinced that public lowlevel api should not contain **a framework** as a dependency, same as we dont include typescipt or webpack

**Comment by szymonlesisz** [comment]
imho the best temporary solution is to remove this file completely from the connect package since it's not used anyway and it's confusing to have it there.

it could be easily restored (and moved to proper place) once we start to implement real device communication in `suite-native`

**Comment by mroz22** [comment]
trezor-connect absolutely should not have react-native as a dependency. and it does not need it now. it is already possible to pass transport interface instead

**Comment by matejkriz** [comment]
closing in favor of https://github.com/trezor/trezor-suite/pull/8848
